Investigate issues reported by flake8-bugbear

### Summary

The flake8-bugbear plug (not used in MI be default) flags up several issues in the code. They might not all be bugs, but they need checking. 

For example:
`./mantidimaging/gui/widgets/palette_changer/view.py:16:62: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.`

The default argument to PaletteChangerView() would get stored as a member in the presenter. But it is looks like it is never modified. So not an actual bug, but using None could prevent it becoming a bug later.

### Steps To Reproduce

```
pip install flake8-bugbear
flake8
```

### Context

Current main

### Failure Logs

Full list

```
./setup.py:139:19: B007 Loop control variable 'dirnames' not used within the loop body. If this is intended, start the name with an underscore.
./mantidimaging/gui/test/gui_system_loading_test.py:49:21: B007 Loop control variable 'x' not used within the loop body. If this is intended, start the name with an underscore.
./mantidimaging/gui/windows/main/test/view_test.py:150:59: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/main/test/view_test.py:151:56: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/main/test/presenter_test.py:447:9: B017 assertRaises(Exception): should be considered evil. It can lead to your test passing even if the code being tested is never executed due to a typo. Either assert for a more specific exception (builtin or custom), use assertRaisesRegex, or use the context manager form of assertRaises.
./mantidimaging/gui/windows/main/test/strictdataset_test.py:50:9: B017 assertRaises(Exception): should be considered evil. It can lead to your test passing even if the code being tested is never executed due to a typo. Either assert for a more specific exception (builtin or custom), use assertRaisesRegex, or use the context manager form of assertRaises.
./mantidimaging/gui/windows/operations/presenter.py:79:9: B007 Loop control variable 'k' not used within the loop body. If this is intended, start the name with an underscore.
./mantidimaging/gui/windows/operations/test/presenter_test.py:99:61: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:100:65: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:101:64: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:121:64: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:121:97: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:141:67: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:142:71: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:143:70: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:167:63: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:168:67: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:169:66: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:189:92: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:190:97: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:278:80: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/operations/test/presenter_test.py:279:85: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/stack_choice/tests/view_test.py:202:75: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/stack_choice/tests/view_test.py:202:112: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/stack_choice/tests/presenter_test.py:56:56: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/windows/stack_choice/tests/presenter_test.py:56:83: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/gui/widgets/palette_changer/view.py:16:62: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
./mantidimaging/core/utility/test/imat_log_file_parser_test.py:63:9: B011 Do not call assert False since python -O removes these calls. Instead callers should raise AssertionError().
./mantidimaging/core/utility/test/imat_log_file_parser_test.py:67:9: B011 Do not call assert False since python -O removes these calls. Instead callers should raise AssertionError().
./mantidimaging/core/parallel/test/shared_test.py:34:13: B007 Loop control variable 'n' not used within the loop body. If this is intended, start the name with an underscore.
./mantidimaging/core/data/reconlist.py:12:49: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
./mantidimaging/core/data/dataset.py:67:51: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
./mantidimaging/core/data/test/image_stack_test.py:215:9: B017 assertRaises(Exception): should be considered evil. It can lead to your test passing even if the code being tested is never executed due to a typo. Either assert for a more specific exception (builtin or custom), use assertRaisesRegex, or use the context manager form of assertRaises.
./mantidimaging/core/gpu/utility.py:88:30: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.
./mantidimaging/core/operations/loader.py:12:26: B007 Loop control variable 'is_pkg' not used within the loop body. If this is intended, start the name with an underscore.
./mantidimaging/core/operations/roi_normalisation/roi_normalisation.py:44:47: B008 Do not perform function calls in argument defaults.  The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call.  If this is intended, assign the function call to a module-level variable and use that variable as a default value.
./mantidimaging/core/operations/rescale/rescale_test.py:110:9: B011 Do not call assert False since python -O removes these calls. Instead callers should raise AssertionError().
./mantidimaging/test_helpers/start_qapplication.py:149:9: B010 Do not call setattr with a constant attribute value, it is not any safer than normal property access.
./mantidimaging/test_helpers/start_qapplication.py:153:9: B010 Do not call setattr with a constant attribute value, it is not any safer than normal property access.
./mantidimaging/test_helpers/start_qapplication.py:157:9: B010 Do not call setattr with a constant attribute value, it is not any safer than normal property access.
./mantidimaging/test_helpers/start_qapplication.py:161:9: B010 Do not call setattr with a constant attribute value, it is not any safer than normal property access.


```


### Screenshot(s)

<!--- [Optional] Include screenshots that illustrate the problem. -->

--- a/docs/release_notes/next.rst
+++ b/docs/release_notes/next.rst
@@ -15,3 +15,5 @@ Fixes
 
 Developer Changes
 -----------------
+
+- #1472 : Investigate issues reported by flake8-bugbear

--- a/mantidimaging/core/data/dataset.py
+++ b/mantidimaging/core/data/dataset.py
@@ -64,8 +64,9 @@ class BaseDataset:
 
 
 class MixedDataset(BaseDataset):
-    def __init__(self, stacks: List[ImageStack] = [], name: str = ""):
+    def __init__(self, stacks: Optional[List[ImageStack]] = None, name: str = ""):
         super().__init__(name=name)
+        stacks = [] if stacks is None else stacks
         self._stacks = stacks
 
     @property

--- a/mantidimaging/core/data/reconlist.py
+++ b/mantidimaging/core/data/reconlist.py
@@ -3,13 +3,14 @@
 
 import uuid
 from collections import UserList
-from typing import List
+from typing import List, Optional
 
 from mantidimaging.core.data import ImageStack
 
 
 class ReconList(UserList):
-    def __init__(self, data: List[ImageStack] = []):
+    def __init__(self, data: Optional[List[ImageStack]] = None):
+        data = [] if data is None else data
         super().__init__(data)
         self._id: uuid.UUID = uuid.uuid4()
 

--- a/mantidimaging/core/data/test/image_stack_test.py
+++ b/mantidimaging/core/data/test/image_stack_test.py
@@ -212,7 +212,7 @@ class ImageStackTest(unittest.TestCase):
         self.assertRaises(ValueError, lambda a, b: a == b, data_images, 1.0)
 
     def test_cant_change_id(self):
-        with self.assertRaises(Exception):
+        with self.assertRaisesRegex(Exception, "can't set attribute"):
             generate_images().id = "id"
 
     def test_default_name(self):

--- a/mantidimaging/core/gpu/utility.py
+++ b/mantidimaging/core/gpu/utility.py
@@ -85,7 +85,7 @@ def _load_cuda_kernel(dtype):
     return cuda_kernel
 
 
-def _free_memory_pool(arrays=[]):
+def _free_memory_pool(arrays=None):
     """
     Delete any given GPU arrays and instruct the memory pool to free unused blocks.
     """

--- a/mantidimaging/core/operations/loader.py
+++ b/mantidimaging/core/operations/loader.py
@@ -9,7 +9,7 @@ from importlib.abc import Loader
 from mantidimaging.core.operations.base_filter import BaseFilter
 
 MODULES_OPERATIONS = {}
-for finder, module_name, is_pkg in pkgutil.walk_packages([os.path.dirname(__file__)]):
+for finder, module_name, _ in pkgutil.walk_packages([os.path.dirname(__file__)]):
     assert isinstance(finder, FileFinder)
     spec = finder.find_spec(module_name)
     assert isinstance(spec, ModuleSpec)

--- a/mantidimaging/core/operations/rescale/rescale_test.py
+++ b/mantidimaging/core/operations/rescale/rescale_test.py
@@ -85,7 +85,7 @@ def test_scale_single_image_bad_offset():
     except ValueError:
         pass
     except Exception as e:
-        assert False, f"Unexpected exception was triggered: {e}"
+        AssertionError(f"Unexpected exception was triggered: {e}")
 
 
 @pytest.mark.parametrize('value', [255.0, 65535.0, 2147483647.0])

--- a/mantidimaging/core/operations/roi_normalisation/roi_normalisation.py
+++ b/mantidimaging/core/operations/roi_normalisation/roi_normalisation.py
@@ -23,6 +23,9 @@ def modes() -> List[str]:
     return ['Stack Average', 'Flat Field']
 
 
+DEFAULT_NORMALISATION_MODE = modes()[0]
+
+
 class RoiNormalisationFilter(BaseFilter):
     """Normalises the image data by using the average value in a no-sample (air) region of interest (ROI) of the
     image. This scaling operation allows to account for beam fluctuations and different exposure times of
@@ -41,7 +44,7 @@ class RoiNormalisationFilter(BaseFilter):
     @staticmethod
     def filter_func(images: ImageStack,
                     region_of_interest: SensibleROI = None,
-                    normalisation_mode: str = modes()[0],
+                    normalisation_mode: str = DEFAULT_NORMALISATION_MODE,
                     flat_field: Optional[ImageStack] = None,
                     progress=None):
         """Normalise by beam intensity.

--- a/mantidimaging/core/operations/test/operations_test.py
+++ b/mantidimaging/core/operations/test/operations_test.py
@@ -54,7 +54,7 @@ class OperationsTest(unittest.TestCase):
             images = th.generate_images()
             if filter_name == "Monitor Normalisation":
                 counts = Counts(numpy.ones(images.num_images))
-                images._log_file = mock.Mock(counts=lambda: counts)
+                images._log_file = mock.Mock(counts=lambda counts=counts: counts)
 
             returned = filter.filter_func(images, **filter_args)
             self.assertEqual(id(images), id(returned))

--- a/mantidimaging/core/parallel/test/shared_test.py
+++ b/mantidimaging/core/parallel/test/shared_test.py
@@ -31,7 +31,7 @@ class SharedTest(unittest.TestCase):
 
     def _create_array_list(self, num_arrays, has_shared_mem):
         array_list = []
-        for n in range(num_arrays):
+        for _ in range(num_arrays):
             mock_array = mock.Mock()
             mock_array.has_shared_memory = has_shared_mem
             mock_array.array_proxy = SharedArrayProxy(None, (2, 2), 'float32') if has_shared_mem else mock.Mock()

--- a/mantidimaging/core/utility/test/imat_log_file_parser_test.py
+++ b/mantidimaging/core/utility/test/imat_log_file_parser_test.py
@@ -60,11 +60,11 @@ def test_counts_compare():
 def assert_raises(exc_type, callable, *args, **kwargs):
     try:
         callable(*args, **kwargs)
-        assert False, "Did not raise"
+        raise AssertionError("Did not raise")
     except exc_type:
         return
     except Exception:
-        assert False, "Did not raise expected exception."
+        raise AssertionError("Did not raise expected exception.")
 
 
 @pytest.mark.parametrize('test_input', [TXT_LOG_FILE, CSV_LOG_FILE])

--- a/mantidimaging/gui/test/gui_system_loading_test.py
+++ b/mantidimaging/gui/test/gui_system_loading_test.py
@@ -46,7 +46,7 @@ class TestGuiSystemLoading(GuiSystemBase):
         cls._wait_for_widget_visible(DatasetSelectorDialog)
         for widget in cls.app.topLevelWidgets():
             if isinstance(widget, DatasetSelectorDialog):
-                for x in range(20):
+                for _ in range(20):
                     QApplication.processEvents(QEventLoop.ProcessEventsFlag.AllEvents, SHORT_DELAY)
                     if widget.dataset_selector_widget.currentText():
                         break

--- a/mantidimaging/gui/test/gui_system_reconstruction_test.py
+++ b/mantidimaging/gui/test/gui_system_reconstruction_test.py
@@ -44,7 +44,7 @@ class TestGuiSystemReconstruction(GuiSystemBase):
 
     def test_minimise(self):
         for i in range(2, 6):
-            QTimer.singleShot(SHORT_DELAY, lambda: self._click_InputDialog(set_int=i))
+            QTimer.singleShot(SHORT_DELAY, lambda i=i: self._click_InputDialog(set_int=i))
             QTest.mouseClick(self.recon_window.minimiseBtn, Qt.MouseButton.LeftButton)
 
             QTest.qWait(SHORT_DELAY)

--- a/mantidimaging/gui/widgets/palette_changer/view.py
+++ b/mantidimaging/gui/widgets/palette_changer/view.py
@@ -13,8 +13,9 @@ class PaletteChangerView(BaseDialogView):
     algorithmComboBox: QComboBox
     colourMapComboBox: QComboBox
 
-    def __init__(self, parent, main_hist, image, other_hists=[], recon_mode=False):
+    def __init__(self, parent, main_hist, image, other_hists=None, recon_mode=False):
         super().__init__(parent, "gui/ui/palette_changer.ui")
+        other_hists = [] if other_hists is None else other_hists
         self.presenter = PaletteChangerPresenter(self, other_hists, main_hist, image, recon_mode)
         self.accepted.connect(self.presenter.change_colour_palette)
         self.colourMapComboBox.setCurrentText("spectrum")

--- a/mantidimaging/gui/windows/main/test/presenter_test.py
+++ b/mantidimaging/gui/windows/main/test/presenter_test.py
@@ -449,8 +449,9 @@ class MainWindowPresenterTest(unittest.TestCase):
     def test_focus_tab_with_id_not_found(self):
         self.model.image_ids = []
         self.model.datasets = []
+        self.model.recon_list_ids = []
 
-        with self.assertRaises(Exception):
+        with self.assertRaises(RuntimeError):
             self.presenter._restore_and_focus_tab(stack_id="not-in-the-stacks-dict")
 
     def test_focus_tab_with_id_in_dataset(self):

--- a/mantidimaging/gui/windows/main/test/strictdataset_test.py
+++ b/mantidimaging/gui/windows/main/test/strictdataset_test.py
@@ -47,7 +47,7 @@ class StrictDatasetTest(unittest.TestCase):
         self.assertIsNone(dataset.dark_after)
 
     def test_cant_change_dataset_id(self):
-        with self.assertRaises(Exception):
+        with self.assertRaisesRegex(Exception, "set attribute"):
             self.strict_dataset.id = "id"
 
     def test_set_flat_before(self):

--- a/mantidimaging/gui/windows/main/test/view_test.py
+++ b/mantidimaging/gui/windows/main/test/view_test.py
@@ -2,6 +2,7 @@
 # SPDX - License - Identifier: GPL-3.0-or-later
 
 import unittest
+from typing import Optional
 from unittest.mock import DEFAULT, Mock
 from uuid import uuid4
 
@@ -147,8 +148,12 @@ class MainWindowViewTest(unittest.TestCase):
     @mock.patch("mantidimaging.gui.windows.main.view.StackVisualiserView")
     def test_create_stack_window(self,
                                  mock_sv: mock.Mock,
-                                 setCentralWidget: Mock = Mock(),
-                                 addDockWidget: Mock = Mock()):
+                                 setCentralWidget: Optional[mock.Mock] = None,
+                                 addDockWidget: Optional[mock.Mock] = None):
+
+        setCentralWidget = mock.Mock() if setCentralWidget is None else setCentralWidget
+        addDockWidget = mock.Mock() if addDockWidget is None else addDockWidget
+
         images = generate_images()
         position = "test_position"
         floating = False

--- a/mantidimaging/gui/windows/main/test/view_test.py
+++ b/mantidimaging/gui/windows/main/test/view_test.py
@@ -2,7 +2,6 @@
 # SPDX - License - Identifier: GPL-3.0-or-later
 
 import unittest
-from typing import Optional
 from unittest.mock import DEFAULT, Mock
 from uuid import uuid4
 
@@ -146,13 +145,7 @@ class MainWindowViewTest(unittest.TestCase):
                          setCentralWidget=DEFAULT,
                          addDockWidget=DEFAULT)
     @mock.patch("mantidimaging.gui.windows.main.view.StackVisualiserView")
-    def test_create_stack_window(self,
-                                 mock_sv: mock.Mock,
-                                 setCentralWidget: Optional[mock.Mock] = None,
-                                 addDockWidget: Optional[mock.Mock] = None):
-
-        setCentralWidget = mock.Mock() if setCentralWidget is None else setCentralWidget
-        addDockWidget = mock.Mock() if addDockWidget is None else addDockWidget
+    def test_create_stack_window(self, mock_sv: mock.Mock, setCentralWidget: mock.Mock, addDockWidget: mock.Mock):
 
         images = generate_images()
         position = "test_position"

--- a/mantidimaging/gui/windows/operations/presenter.py
+++ b/mantidimaging/gui/windows/operations/presenter.py
@@ -80,7 +80,7 @@ def _group_consecutive_values(slices: List[int]) -> List[str]:
     :return: A list of strings for the index ranges e.g. 1-5, 7-20, etc
     """
     ranges = []
-    for k, iterable in groupby(enumerate(slices), sub):
+    for _, iterable in groupby(enumerate(slices), sub):
         rng = list(iterable)
         if len(rng) == 1:
             s = str(rng[0][1])

--- a/mantidimaging/gui/windows/operations/test/presenter_test.py
+++ b/mantidimaging/gui/windows/operations/test/presenter_test.py
@@ -5,7 +5,7 @@ import unittest
 import numpy as np
 import numpy.testing as npt
 from functools import partial
-from typing import List
+from typing import List, Optional
 
 from unittest import mock
 from unittest.mock import DEFAULT, Mock
@@ -99,12 +99,16 @@ class FiltersWindowPresenterTest(unittest.TestCase):
                          _wait_for_stack_choice=DEFAULT,
                          _do_apply_filter_sync=DEFAULT)
     def test_post_filter_success(self,
-                                 do_update_previews: Mock = Mock(),
-                                 _wait_for_stack_choice: Mock = Mock(),
-                                 _do_apply_filter_sync: Mock = Mock()):
+                                 do_update_previews: Optional[Mock] = None,
+                                 _wait_for_stack_choice: Optional[Mock] = None,
+                                 _do_apply_filter_sync: Optional[Mock] = None):
         """
         Tests when the operation has applied successfully.
         """
+        do_update_previews = Mock() if do_update_previews is None else do_update_previews
+        _wait_for_stack_choice = Mock() if _wait_for_stack_choice is None else _wait_for_stack_choice
+        _do_apply_filter_sync = Mock() if _do_apply_filter_sync is None else _do_apply_filter_sync
+
         self.presenter.view.safeApply.isChecked.return_value = False
         mock_task = mock.Mock()
         mock_task.error = None

--- a/mantidimaging/gui/windows/operations/test/presenter_test.py
+++ b/mantidimaging/gui/windows/operations/test/presenter_test.py
@@ -21,6 +21,12 @@ from mantidimaging.test_helpers.unit_test_helper import assert_called_once_with,
 from mantidimaging.core.data import ImageStack
 
 
+def _create_mock_if_none(mocked_object) -> Mock:
+    if mocked_object is None:
+        return Mock()
+    return mocked_object
+
+
 class FiltersWindowPresenterTest(unittest.TestCase):
     def setUp(self) -> None:
         self.main_window = mock.create_autospec(MainWindowView)
@@ -105,9 +111,9 @@ class FiltersWindowPresenterTest(unittest.TestCase):
         """
         Tests when the operation has applied successfully.
         """
-        do_update_previews = Mock() if do_update_previews is None else do_update_previews
-        _wait_for_stack_choice = Mock() if _wait_for_stack_choice is None else _wait_for_stack_choice
-        _do_apply_filter_sync = Mock() if _do_apply_filter_sync is None else _do_apply_filter_sync
+        do_update_previews = _create_mock_if_none(do_update_previews)
+        _wait_for_stack_choice = _create_mock_if_none(_wait_for_stack_choice)
+        _do_apply_filter_sync = _create_mock_if_none(_do_apply_filter_sync)
 
         self.presenter.view.safeApply.isChecked.return_value = False
         mock_task = mock.Mock()
@@ -125,10 +131,13 @@ class FiltersWindowPresenterTest(unittest.TestCase):
     @mock.patch.multiple('mantidimaging.gui.windows.operations.presenter.FiltersWindowPresenter',
                          do_update_previews=DEFAULT,
                          _do_apply_filter=DEFAULT)
-    def test_post_filter_fail(self, do_update_previews: Mock = Mock(), _do_apply_filter: Mock = Mock()):
+    def test_post_filter_fail(self, do_update_previews: Optional[Mock] = None, _do_apply_filter: Optional[Mock] = None):
         """
         Tests when the operation has encountered an error.
         """
+        do_update_previews = _create_mock_if_none(do_update_previews)
+        _do_apply_filter = _create_mock_if_none(_do_apply_filter)
+
         self.presenter.view.safeApply.isChecked.return_value = False
         self.presenter.view.show_error_dialog = mock.Mock()  # type: ignore
         self.presenter.main_window.presenter = mock.Mock()
@@ -145,12 +154,16 @@ class FiltersWindowPresenterTest(unittest.TestCase):
                          _wait_for_stack_choice=DEFAULT,
                          _do_apply_filter_sync=DEFAULT)
     def test_post_filter_keep_original(self,
-                                       do_update_previews: Mock = Mock(),
-                                       _wait_for_stack_choice: Mock = Mock(),
-                                       _do_apply_filter_sync: Mock = Mock()):
+                                       do_update_previews: Optional[Mock] = None,
+                                       _wait_for_stack_choice: Optional[Mock] = None,
+                                       _do_apply_filter_sync: Optional[Mock] = None):
         """
         Tests when the operation has applied successfully, but user choose to keep original
         """
+        do_update_previews = _create_mock_if_none(do_update_previews)
+        _wait_for_stack_choice = _create_mock_if_none(_wait_for_stack_choice)
+        _do_apply_filter_sync = _create_mock_if_none(_do_apply_filter_sync)
+
         self.presenter.view.safeApply.isChecked.return_value = True
         mock_task = mock.Mock()
         mock_task.error = None
@@ -171,12 +184,16 @@ class FiltersWindowPresenterTest(unittest.TestCase):
                          _wait_for_stack_choice=DEFAULT,
                          _do_apply_filter_sync=DEFAULT)
     def test_post_filter_exception(self,
-                                   do_update_previews: Mock = Mock(),
-                                   _wait_for_stack_choice: Mock = Mock(),
-                                   _do_apply_filter_sync: Mock = Mock()):
+                                   do_update_previews: Optional[Mock] = None,
+                                   _wait_for_stack_choice: Optional[Mock] = None,
+                                   _do_apply_filter_sync: Optional[Mock] = None):
         """
         Tests that an exception in _post_filter() does not prevent filter_is_running being set to False
         """
+        do_update_previews = _create_mock_if_none(do_update_previews)
+        _wait_for_stack_choice = _create_mock_if_none(_wait_for_stack_choice)
+        _do_apply_filter_sync = _create_mock_if_none(_do_apply_filter_sync)
+
         self.presenter.view.safeApply.isChecked.return_value = False
         mock_task = mock.Mock()
         mock_task.error = None
@@ -193,13 +210,16 @@ class FiltersWindowPresenterTest(unittest.TestCase):
         _do_apply_filter_sync=DEFAULT,
     )
     def test_images_with_180_deg_proj_calls_filter_on_the_180_deg(self,
-                                                                  _do_apply_filter: Mock = Mock(),
-                                                                  _do_apply_filter_sync: Mock = Mock()):
+                                                                  _do_apply_filter: Optional[Mock] = None,
+                                                                  _do_apply_filter_sync: Optional[Mock] = None):
         """
         Test that when an `ImageStack` stack is encountered which also has a
         180deg projection stack reference, that 180deg stack is also processed
         with the same operation, to ensure consistency between the two images
         """
+        _do_apply_filter = _create_mock_if_none(_do_apply_filter)
+        _do_apply_filter_sync = _create_mock_if_none(_do_apply_filter_sync)
+
         self.presenter.view.safeApply.isChecked.return_value = False
         self.presenter.applying_to_all = False
         mock_stack = mock.MagicMock()
@@ -329,8 +349,11 @@ class FiltersWindowPresenterTest(unittest.TestCase):
     @mock.patch('mantidimaging.gui.windows.operations.presenter.StackChoicePresenter')
     def test_safe_apply_starts_stack_choice_presenter(self,
                                                       stack_choice_presenter: Mock,
-                                                      _do_apply_filter: Mock = Mock(),
-                                                      _do_apply_filter_sync: Mock = Mock()):
+                                                      _do_apply_filter: Optional[Mock] = None,
+                                                      _do_apply_filter_sync: Optional[Mock] = None):
+        _do_apply_filter = _create_mock_if_none(_do_apply_filter)
+        _do_apply_filter_sync = _create_mock_if_none(_do_apply_filter_sync)
+
         task = Mock()
         task.error = None
 

--- a/mantidimaging/gui/windows/operations/test/presenter_test.py
+++ b/mantidimaging/gui/windows/operations/test/presenter_test.py
@@ -5,7 +5,7 @@ import unittest
 import numpy as np
 import numpy.testing as npt
 from functools import partial
-from typing import List, Optional
+from typing import List
 
 from unittest import mock
 from unittest.mock import DEFAULT, Mock
@@ -19,12 +19,6 @@ from mantidimaging.gui.windows.operations.presenter import REPEAT_FLAT_FIELDING_
     _group_consecutive_values, FLAT_FIELD_REGION
 from mantidimaging.test_helpers.unit_test_helper import assert_called_once_with, generate_images
 from mantidimaging.core.data import ImageStack
-
-
-def _create_mock_if_none(mocked_object) -> Mock:
-    if mocked_object is None:
-        return Mock()
-    return mocked_object
 
 
 class FiltersWindowPresenterTest(unittest.TestCase):
@@ -322,13 +316,8 @@ class FiltersWindowPresenterTest(unittest.TestCase):
         _do_apply_filter_sync=DEFAULT,
     )
     @mock.patch('mantidimaging.gui.windows.operations.presenter.StackChoicePresenter')
-    def test_safe_apply_starts_stack_choice_presenter(self,
-                                                      stack_choice_presenter: Mock,
-                                                      _do_apply_filter: Optional[Mock] = None,
-                                                      _do_apply_filter_sync: Optional[Mock] = None):
-        _do_apply_filter = _create_mock_if_none(_do_apply_filter)
-        _do_apply_filter_sync = _create_mock_if_none(_do_apply_filter_sync)
-
+    def test_safe_apply_starts_stack_choice_presenter(self, stack_choice_presenter: Mock, _do_apply_filter: Mock,
+                                                      _do_apply_filter_sync: Mock):
         task = Mock()
         task.error = None
 

--- a/mantidimaging/gui/windows/operations/test/presenter_test.py
+++ b/mantidimaging/gui/windows/operations/test/presenter_test.py
@@ -104,17 +104,11 @@ class FiltersWindowPresenterTest(unittest.TestCase):
                          do_update_previews=DEFAULT,
                          _wait_for_stack_choice=DEFAULT,
                          _do_apply_filter_sync=DEFAULT)
-    def test_post_filter_success(self,
-                                 do_update_previews: Optional[Mock] = None,
-                                 _wait_for_stack_choice: Optional[Mock] = None,
-                                 _do_apply_filter_sync: Optional[Mock] = None):
+    def test_post_filter_success(self, do_update_previews: Mock, _wait_for_stack_choice: Mock,
+                                 _do_apply_filter_sync: Mock):
         """
         Tests when the operation has applied successfully.
         """
-        do_update_previews = _create_mock_if_none(do_update_previews)
-        _wait_for_stack_choice = _create_mock_if_none(_wait_for_stack_choice)
-        _do_apply_filter_sync = _create_mock_if_none(_do_apply_filter_sync)
-
         self.presenter.view.safeApply.isChecked.return_value = False
         mock_task = mock.Mock()
         mock_task.error = None
@@ -131,13 +125,10 @@ class FiltersWindowPresenterTest(unittest.TestCase):
     @mock.patch.multiple('mantidimaging.gui.windows.operations.presenter.FiltersWindowPresenter',
                          do_update_previews=DEFAULT,
                          _do_apply_filter=DEFAULT)
-    def test_post_filter_fail(self, do_update_previews: Optional[Mock] = None, _do_apply_filter: Optional[Mock] = None):
+    def test_post_filter_fail(self, do_update_previews: Mock, _do_apply_filter: Mock):
         """
         Tests when the operation has encountered an error.
         """
-        do_update_previews = _create_mock_if_none(do_update_previews)
-        _do_apply_filter = _create_mock_if_none(_do_apply_filter)
-
         self.presenter.view.safeApply.isChecked.return_value = False
         self.presenter.view.show_error_dialog = mock.Mock()  # type: ignore
         self.presenter.main_window.presenter = mock.Mock()
@@ -153,17 +144,11 @@ class FiltersWindowPresenterTest(unittest.TestCase):
                          do_update_previews=DEFAULT,
                          _wait_for_stack_choice=DEFAULT,
                          _do_apply_filter_sync=DEFAULT)
-    def test_post_filter_keep_original(self,
-                                       do_update_previews: Optional[Mock] = None,
-                                       _wait_for_stack_choice: Optional[Mock] = None,
-                                       _do_apply_filter_sync: Optional[Mock] = None):
+    def test_post_filter_keep_original(self, do_update_previews: Mock, _wait_for_stack_choice: Mock,
+                                       _do_apply_filter_sync: Mock):
         """
         Tests when the operation has applied successfully, but user choose to keep original
         """
-        do_update_previews = _create_mock_if_none(do_update_previews)
-        _wait_for_stack_choice = _create_mock_if_none(_wait_for_stack_choice)
-        _do_apply_filter_sync = _create_mock_if_none(_do_apply_filter_sync)
-
         self.presenter.view.safeApply.isChecked.return_value = True
         mock_task = mock.Mock()
         mock_task.error = None
@@ -183,17 +168,11 @@ class FiltersWindowPresenterTest(unittest.TestCase):
                          do_update_previews=DEFAULT,
                          _wait_for_stack_choice=DEFAULT,
                          _do_apply_filter_sync=DEFAULT)
-    def test_post_filter_exception(self,
-                                   do_update_previews: Optional[Mock] = None,
-                                   _wait_for_stack_choice: Optional[Mock] = None,
-                                   _do_apply_filter_sync: Optional[Mock] = None):
+    def test_post_filter_exception(self, do_update_previews: Mock, _wait_for_stack_choice: Mock,
+                                   _do_apply_filter_sync: Mock):
         """
         Tests that an exception in _post_filter() does not prevent filter_is_running being set to False
         """
-        do_update_previews = _create_mock_if_none(do_update_previews)
-        _wait_for_stack_choice = _create_mock_if_none(_wait_for_stack_choice)
-        _do_apply_filter_sync = _create_mock_if_none(_do_apply_filter_sync)
-
         self.presenter.view.safeApply.isChecked.return_value = False
         mock_task = mock.Mock()
         mock_task.error = None
@@ -209,17 +188,13 @@ class FiltersWindowPresenterTest(unittest.TestCase):
         _do_apply_filter=DEFAULT,
         _do_apply_filter_sync=DEFAULT,
     )
-    def test_images_with_180_deg_proj_calls_filter_on_the_180_deg(self,
-                                                                  _do_apply_filter: Optional[Mock] = None,
-                                                                  _do_apply_filter_sync: Optional[Mock] = None):
+    def test_images_with_180_deg_proj_calls_filter_on_the_180_deg(self, _do_apply_filter: Mock,
+                                                                  _do_apply_filter_sync: Mock):
         """
         Test that when an `ImageStack` stack is encountered which also has a
         180deg projection stack reference, that 180deg stack is also processed
         with the same operation, to ensure consistency between the two images
         """
-        _do_apply_filter = _create_mock_if_none(_do_apply_filter)
-        _do_apply_filter_sync = _create_mock_if_none(_do_apply_filter_sync)
-
         self.presenter.view.safeApply.isChecked.return_value = False
         self.presenter.applying_to_all = False
         mock_stack = mock.MagicMock()

--- a/mantidimaging/gui/windows/stack_choice/tests/presenter_test.py
+++ b/mantidimaging/gui/windows/stack_choice/tests/presenter_test.py
@@ -2,7 +2,6 @@
 # SPDX - License - Identifier: GPL-3.0-or-later
 
 import unittest
-from typing import Optional
 from unittest import mock
 from unittest.mock import DEFAULT, MagicMock, Mock, patch
 from uuid import uuid4
@@ -54,7 +53,7 @@ class StackChoicePresenterTest(unittest.TestCase):
     @patch.multiple("mantidimaging.gui.windows.stack_choice.presenter.StackChoicePresenter",
                     do_reapply_original_data=MagicMock(side_effect=RuntimeError),
                     show_error=DEFAULT)
-    def test_notify_handles_exceptions(self, _: Optional[Mock] = None, show_error: Optional[Mock] = None):
+    def test_notify_handles_exceptions(self, _: Mock, show_error: Mock):
         show_error = mock.Mock() if show_error is None else show_error
         self.p.notify(Notification.CHOOSE_ORIGINAL)
 

--- a/mantidimaging/gui/windows/stack_choice/tests/presenter_test.py
+++ b/mantidimaging/gui/windows/stack_choice/tests/presenter_test.py
@@ -2,6 +2,7 @@
 # SPDX - License - Identifier: GPL-3.0-or-later
 
 import unittest
+from typing import Optional
 from unittest import mock
 from unittest.mock import DEFAULT, MagicMock, Mock, patch
 from uuid import uuid4
@@ -53,7 +54,8 @@ class StackChoicePresenterTest(unittest.TestCase):
     @patch.multiple("mantidimaging.gui.windows.stack_choice.presenter.StackChoicePresenter",
                     do_reapply_original_data=MagicMock(side_effect=RuntimeError),
                     show_error=DEFAULT)
-    def test_notify_handles_exceptions(self, _: Mock = Mock(), show_error: Mock = Mock()):
+    def test_notify_handles_exceptions(self, _: Optional[Mock] = None, show_error: Optional[Mock] = None):
+        show_error = mock.Mock() if show_error is None else show_error
         self.p.notify(Notification.CHOOSE_ORIGINAL)
 
         show_error.assert_called_once()

--- a/mantidimaging/gui/windows/stack_choice/tests/presenter_test.py
+++ b/mantidimaging/gui/windows/stack_choice/tests/presenter_test.py
@@ -53,8 +53,7 @@ class StackChoicePresenterTest(unittest.TestCase):
     @patch.multiple("mantidimaging.gui.windows.stack_choice.presenter.StackChoicePresenter",
                     do_reapply_original_data=MagicMock(side_effect=RuntimeError),
                     show_error=DEFAULT)
-    def test_notify_handles_exceptions(self, _: Mock, show_error: Mock):
-        show_error = mock.Mock() if show_error is None else show_error
+    def test_notify_handles_exceptions(self, _: Mock = Mock(), show_error: Mock = Mock()):
         self.p.notify(Notification.CHOOSE_ORIGINAL)
 
         show_error.assert_called_once()

--- a/mantidimaging/gui/windows/stack_choice/tests/presenter_test.py
+++ b/mantidimaging/gui/windows/stack_choice/tests/presenter_test.py
@@ -53,7 +53,7 @@ class StackChoicePresenterTest(unittest.TestCase):
     @patch.multiple("mantidimaging.gui.windows.stack_choice.presenter.StackChoicePresenter",
                     do_reapply_original_data=MagicMock(side_effect=RuntimeError),
                     show_error=DEFAULT)
-    def test_notify_handles_exceptions(self, _: Mock = Mock(), show_error: Mock = Mock()):
+    def test_notify_handles_exceptions(self, show_error: Mock):
         self.p.notify(Notification.CHOOSE_ORIGINAL)
 
         show_error.assert_called_once()

--- a/mantidimaging/gui/windows/stack_choice/tests/view_test.py
+++ b/mantidimaging/gui/windows/stack_choice/tests/view_test.py
@@ -1,6 +1,7 @@
 # Copyright (C) 2022 ISIS Rutherford Appleton Laboratory UKRI
 # SPDX - License - Identifier: GPL-3.0-or-later
 import unittest
+from typing import Optional
 from unittest import mock
 from unittest.mock import DEFAULT, Mock, patch
 from uuid import uuid4
@@ -199,7 +200,12 @@ class StackChoiceViewTest(unittest.TestCase):
     @patch.multiple('mantidimaging.gui.windows.stack_choice.view.StackChoiceView',
                     _set_from_old_to_new=DEFAULT,
                     _set_from_new_to_old=DEFAULT)
-    def test_connect_histogram_changes(self, _set_from_old_to_new: Mock = Mock(), _set_from_new_to_old: Mock = Mock()):
+    def test_connect_histogram_changes(self,
+                                       _set_from_old_to_new: Optional[Mock] = None,
+                                       _set_from_new_to_old: Optional[Mock] = None):
+        _set_from_old_to_new = mock.Mock() if _set_from_old_to_new is None else _set_from_old_to_new
+        _set_from_new_to_old = mock.Mock() if _set_from_new_to_old is None else _set_from_new_to_old
+
         self.v.connect_histogram_changes()
 
         # check this is called once to set the same range on the new histogram as is currently selected on the new one

--- a/mantidimaging/gui/windows/stack_choice/tests/view_test.py
+++ b/mantidimaging/gui/windows/stack_choice/tests/view_test.py
@@ -1,7 +1,6 @@
 # Copyright (C) 2022 ISIS Rutherford Appleton Laboratory UKRI
 # SPDX - License - Identifier: GPL-3.0-or-later
 import unittest
-from typing import Optional
 from unittest import mock
 from unittest.mock import DEFAULT, Mock, patch
 from uuid import uuid4
@@ -200,11 +199,7 @@ class StackChoiceViewTest(unittest.TestCase):
     @patch.multiple('mantidimaging.gui.windows.stack_choice.view.StackChoiceView',
                     _set_from_old_to_new=DEFAULT,
                     _set_from_new_to_old=DEFAULT)
-    def test_connect_histogram_changes(self,
-                                       _set_from_old_to_new: Optional[Mock] = None,
-                                       _set_from_new_to_old: Optional[Mock] = None):
-        _set_from_old_to_new = mock.Mock() if _set_from_old_to_new is None else _set_from_old_to_new
-        _set_from_new_to_old = mock.Mock() if _set_from_new_to_old is None else _set_from_new_to_old
+    def test_connect_histogram_changes(self, _set_from_old_to_new: Mock, _set_from_new_to_old: Mock):
 
         self.v.connect_histogram_changes()
 

--- a/mantidimaging/test_helpers/start_qapplication.py
+++ b/mantidimaging/test_helpers/start_qapplication.py
@@ -146,17 +146,17 @@ def augment_test_setup_methods(cls, setup=None, teardown=None, setup_class=None,
 
     if setup:
         setup_orig = cls.setUp if hasattr(cls, 'setUp') else do_nothing
-        setattr(cls, 'setUp', setUp)
+        cls.setUp = setUp
 
     if teardown:
         teardown_orig = cls.tearDown if hasattr(cls, 'tearDown') else do_nothing
-        setattr(cls, 'tearDown', tearDown)
+        cls.tearDown = tearDown
 
     if setup_class:
         setup_class_orig = cls.setUpClass if hasattr(cls, 'setUpClass') else do_nothing
-        setattr(cls, 'setUpClass', classmethod(setUpClass))
+        cls.setUpClass = classmethod(setUpClass)
 
     if teardown_class:
         teardown_class_orig = cls.tearDownClass if hasattr(cls, 'tearDownClass') else do_nothing
-        setattr(cls, 'tearDownClass', classmethod(tearDownClass))
+        cls.tearDownClass = classmethod(tearDownClass)
     return cls

--- a/setup.py
+++ b/setup.py
@@ -136,7 +136,7 @@ class CompilePyQtUiFiles(Command):
     @staticmethod
     def find_ui_files():
         matches = []
-        for root, dirnames, filenames in os.walk("./mantidimaging/"):
+        for root, _, filenames in os.walk("./mantidimaging/"):
             for filename in fnmatch.filter(filenames, "*.ui"):
                 matches.append(os.path.join(root, filename))
         return matches

